### ISSUE TITLE
Checkbox indeterminate state has bad attribute causing a11y error

### ISSUE THREAD

**Issue by mellis481** [ant-design/ant-design#51345]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/basic-antd-5-21-5-forked-dqx5w8)

### Steps to reproduce

Run FastPass or other such a11y tool.

### What is expected?

No a11y error.

### What is actually happening?

`aria-conditional-attr` a11y error.

| Environment | Info |
| --- | --- |
| antd | 5.20.0 |
| React | 18 |
| System | Latest |
| Browser | All |

---
It looks like in WCAG 1.1, `aria-checked="mixed"` was the intended way to code a checkbox in indeterminate state so it will be announced correctly by screen readers. https://www.w3.org/TR/wai-aria-practices-1.1/examples/checkbox/checkbox-2/checkbox-2.html. That was implemented in https://github.com/ant-design/ant-design/pull/34862. It looks like that is not correct anymore. Instead, the `indeterminate` attribute should be used. https://dequeuniversity.com/rules/axe/4.8/aria-conditional-attr?application=axeAPI

**Comment by afc163** [comment]
@SpaNb4 Could you help check this issue? https://github.com/ant-design/ant-design/pull/34862
